### PR TITLE
Fix React code style, to surface functional errors (SCP-2222)

### PR DESCRIPTION
**Pull request by eweitz**
This fixes React code style throughout the codebase, to keep static analysis reports of functional errors easily noticeable by engineers.  Previously, almost all React JS files reported many code style errors.   Now, no JS files appear red in an IDE due to style errors.  

This refactoring ensures functional bugs aren't missed due to [alarm fatigue](https://en.wikipedia.org/wiki/Alarm_fatigue) from noisy IDE error reports.  It also keeps linting useful for feature velocity by narrowing where code style rules are disabled.

This satisfies SCP-2222.

### CODE REVIEW

File: app/javascript/components/FilterSlider.js [hunk 1]
```diff
@@ -107,26 +107,26 @@ export default function FilterSlider(props) {
...
         type='number'
         min={min}
         max={max}
         value={inputValues[1]}
-        style={{'width': '60px', 'marginRight': '8px'}}
+        style={{ 'width': '60px', 'marginRight': '8px' }}
       />
-      <select value={unit} onChange={(event) => {updateUnit(event.target.value)}}>
+      <select value={unit} onChange={event => {updateUnit(event.target.value)}}>
         {units.map((unit, i) =>
```
Comment (devonbush) [inline app/javascript/components/FilterSlider.js]: I don't understand where there is a comma added to the end of this line -- is that a stray keystroke, or a style rule?

File: test/js/setup-tests.js [hunk 2]
```diff
@@ -1,12 +1,12 @@
-import { configure } from 'enzyme';
-import Adapter from 'enzyme-adapter-react-16';
+import { configure } from 'enzyme'
+import Adapter from 'enzyme-adapter-react-16'
 
-configure({ adapter: new Adapter() });
+configure({ adapter: new Adapter() })
 
 import {
   setGlobalMockFlag,
```
Comment (devonbush) [inline test/js/setup-tests.js]: I'm not personally a fan of required trailing commas, but don't feel too strongly about it

Comment (eweitz) [reply test/js/setup-tests.js]: Same here.  I just flipped that linter rule to err on trailing commas.

File: app/javascript/components/FilterSlider.js (hunk 1 above)
Comment (eweitz) [reply app/javascript/components/FilterSlider.js]: It was a style rule -- the formatter automatically inserted trailing commas.  I removed that linter rule and this trailing comma.

**Comment by eweitz** [comment]
I'm merging this with only one review after chat with Devon.  Eno also approved style rule refinements in a chat Friday morning.

**Pull request merged**